### ISSUE TITLE
Server Unavailable after period of time

### ISSUE THREAD

**Issue by cjindustries** [datalust/seq-tickets#673]
Hi,

In the latest version of Seq I'm seeing that the server becomes unavailable after a short period of time - I get the error icon in the header. If I then refresh the browser, Chrome tries to connect to https://localhost:5341 rather than over http://.

This has only started happening in the latest version (or maybe the one before) of Seq. To resolve it I have to clear my browsing history, but after a short time it comes back again.

I'm running Seq 4.2.717 on my Windows 10. Browser is Chrome 65.0.3325.146.

Has anyone else seen this? Let me know if there's any more information I can give.

Thanks,
Chris.

**Comment by nblumhardt** [comment]
Hi Chris, thanks for the heads-up.

This sounds like it might be related to HSTS; does your Seq configuration have `api.redirectHttpToHttps` set? You can find this out by running:

```
seq config -k api.redirectHttpToHttps
```

It would also be relevant if your `api.listenUris` setting is HTTPS-only (though this doesn't sound like it would be the case), i.e.:

```
seq config -k api.listenUris
```

Let us know what you find, hopefully a simple config tweak to sort this out.

Best regards,
Nick

**Comment by cjindustries** [comment]
Hello,

Thanks for the pointers, both settings were fine.

However I've found the cause and it's my fault. We recently introduced
NWebSec libraries into our apps and when I run them locally, they configure
HSTS for localhost and stop Seq working correctly. Clearing it
via chrome://net-internals/#hsts allowed Seq to work again.

Sorry for not finding this earlier. Maybe this experience will help others
in future! :)

Thanks,
Chris.
